packages/@aws-cdk/aws-sns/lib/policy.ts

```diff
@@ -22,6 +22,12 @@ export class TopicPolicy extends Resource {
   constructor(scope: Construct, id: string, props: TopicPolicyProps) {
     super(scope, id);
 
+    // statements must be unique, so we use the statement index.
+    // potantially SIDs can change as a result of order change, but this should
+    // not have an impact on the policy evaluation.
+    // https://docs.aws.amazon.com/sns/latest/dg/AccessPolicyLanguage_SpecialInfo.html
+    this.document.autoAssignSids();
```

Comment:
Feels like this should be a constructor property instead of a mutating function.

Reply:
I agree, but I didn't want to break the ctor of `PolicyDocument` (which currently accepts a base document). I'll add it to https://github.com/awslabs/aws-cdk/issues/2344